modules/cart/tests/modules/commerce_cart_big_pipe/src/Form/BigPipeAddToCartForm.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Drupal\commerce_cart_big_pipe\Form;
+
+use Drupal\commerce_cart\Form\AddToCartForm;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Provides a slow to build add to cart form, to test streaming.
+ */
+class BigPipeAddToCartForm extends AddToCartForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    // Sleep between 0.0 and 2.0 seconds.
```

Comment:
According to https://www.drupal.org/node/2886614, there is a chance the forms could be built out of order. To ensure unique form IDs, we ended up using a static counter. If the forms are built out of order, this causes a bug in form submission. My goal is to reproduce the bug and test our assumed fix.

So I thought random times might cause the build to get weird. However by this time the form ID is known... so maybe slowdown happens elsewhere when rendering these formatter placeholders.